app/controllers/account_controllers.rb

```diff
@@ -0,0 +1,46 @@
+
+get '/accounts/login' do
+  erb :login
+end
+
+post '/accounts/login' do
+  user = User.find_by(email: params[:email])
+  db_password = user.password_hash
+  if db_password == params[:password]
+    session[:current_user_id] = user.id
+    redirect ('/accounts/' + user.username)
+  else
+    @error = "Your email or password did not match. Please try again."
+    erb :login
+  end
+end
+
+get '/accounts/:username/edit' do
+  @user = session_current_user
+  erb :edit_account
+end
+
+# bugs with username
+put '/accounts/:username/edit' do
```

Comment:
Restful convention would be `put '/users/:user_id' do` as the update route for a user. Checkout this guide for more info: http://guides.rubyonrails.org/routing.html